stdlib/asm/crypto/fri/helper.masm

```diff
@@ -39,6 +39,10 @@ export.generate_fri_parameters
     repeat.3
         dup mul
     end
+
+    # Save a copy of the domain generator to memory (also used to compute the divisors)
+    dup.0 exec.constants::domain_generator_ptr mem_store
```

Comment:
One question: is this code run before or after constraint evaluation? I'm not 100% sure, but I thought it is run after. If so, we might need to move this somewhere more upfront.

Reply:
I don't know, @Al-Kindi-0 ?

Reply:
Yes, this portion of the code is run after the constraint evaluation phase. Thus, it makes sense to run it during constraint evaluation and store it then for later use by the `helper::generate_fri_parameters` procedure.

Reply:
Maybe we should run it right at the beginning of the verifier procedure? This way, both constraint evaluation code and FRI can assume that this has been computed.

Reply:
Yes, makes sense. The same also holds for the trace domain generator where it is used in computing `g.z` which is itself used in computing the denominators `(x-gz)` for the different `x`'s

Reply:
@Al-Kindi-0 , just to clarify, do you mean these values?

https://github.com/0xPolygonMiden/miden-vm/blob/648941543f3eb2e5a51359daa9ba07f9a26a54ec/stdlib/asm/crypto/stark/deep_queries.masm#L407-L413

Reply:
Indeed, and `gz` is computed here https://github.com/0xPolygonMiden/miden-vm/blob/648941543f3eb2e5a51359daa9ba07f9a26a54ec/stdlib/asm/crypto/fri/helper.masm#L29

Reply:
update: Initially I planned to move this code to `random_coin::init_seed`. I just realized that is not possible because `z` is only withdraw after a few `reseed`s, and that can't be moved.

so instead of moving the computation of `trace_g` to `init_seed`, I'm moving the call to `generate_fri_parameters` up to the point were `z` is defined.

Reply:
Update: Added code to save the trace_g to memory.